mux.go

```diff
@@ -298,7 +298,8 @@ func (m *mux) handleHTTP(srv *http.Server, conn net.Conn, closed <-chan struct{}
 
 func (m *mux) handleGRPC(_ *http.Server, conn net.Conn, closed <-chan struct{}, _ http.Handler) error {
 	select {
-	case m.grpcConns <- conn:
+	case <-m.grpcListener.closed:
```

Comment:
I think we should return an error here indicating that the gRPC listener has been closed.

Reply:
Right. What was I thinking yesterday? :facepalm: 
Though I'll check how caller handles error in case handleGRPC returns non-nil error

Reply:
The error from `handleGRPC` will be logged to `srv.ErrLog` for h2 but ignored for h2c. Will have to pass in `srv.ErrLog` to `mux.withGRPCInsecure` if we want to log the error.

Reply:
Not super important, but sounds like a good idea.